images/hub/jupyterhub_config.py

```diff
@@ -177,6 +178,25 @@ def get_config(key, default=None):
     c.DummyAuthenticator.password = get_config('auth.dummy.password', None)
 elif auth_type == 'tmp':
     c.JupyterHub.authenticator_class = 'tmpauthenticator.TmpAuthenticator'
+elif auth_type == 'custom':
+    # full_class_name looks like "myauthenticator.MyAuthenticator".
+    # To create a docker image with this class availabe, you can just have the
+    # following Dockerifle:
+    #   FROM jupyterhub/k8s-hub:v0.4
+    #   RUN pip3 install myauthenticator
+    full_class_name = get_config('auth.custom.class_name')
+    c.JupyterHub.authenticator_class = full_class_name
+
+    (module, class_name) = full_class_name.split('.')
+    the_module = importlib.import_module(module)
+    klass=getattr(the_module, class_name)
+
+    class_params = get_config('auth.custom.params')
```

Comment:
I suspect there's a much easier way of setting these params. I know that Traitlets (which is what these all use) can take configuration from a JSON file directly - and I'd prefer if we just utilized that rather than do this. @minrk do you know how best to pass a dictionary to authenticator config here?